core/src/main/java/io/kestra/core/runners/ExecutorService.java

```diff
@@ -555,7 +562,7 @@ private Executor handleListeners(Executor executor) {
     }
 
     private Executor handleEnd(Executor executor) {
-        if (executor.getExecution().getState().isTerminated() || executor.getExecution().getState().isPaused()) {
+        if (executor.getExecution().getState().isTerminated() || executor.getExecution().getState().isPaused() || executor.getExecution().getState().isRetrying()) {
```

Comment:
isPaused() check for the RETRYING status so it didn't seems to be useful

Reply:
isPaused shouldnt, was testing leftover